cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/RemoteConfigurator.java

```diff
@@ -0,0 +1,107 @@
+import io.cdap.common.http.HttpMethod;
+import io.cdap.common.http.HttpRequest;
+import io.cdap.common.http.HttpResponse;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.apache.twill.filesystem.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * RemoteConfigurator sends a request to another {@link Configurator} running remotely.
+ */
+public class RemoteConfigurator implements Configurator {
+  private static final Logger LOG = LoggerFactory.getLogger(RemoteConfigurator.class);
+  private static final Gson GSON = new Gson();
+  private final ConfiguratorConfig config;
+
+  private final RemoteClient remoteClientInternal;
+
+  public RemoteConfigurator(ConfiguratorConfig config,
+                            DiscoveryServiceClient discoveryServiceClient) {
+    this.config = config;
+
+    this.remoteClientInternal = new RemoteClient(discoveryServiceClient, Constants.Service.TASK_WORKER,
+                                                 new DefaultHttpRequestConfig(false),
+                                                 String.format("%s", Constants.Gateway.INTERNAL_API_VERSION_3));
+  }
+
+  public RemoteConfigurator(CConfiguration cConf, Id.Namespace appNamespace, Id.Artifact artifactId,
+                            String appClassName, PluginFinder pluginFinder, ClassLoader artifactClassLoader,
+                            String applicationName, String applicationVersion, String configString,
+                            DiscoveryServiceClient discoveryServiceClient,
+                            Location artifactLocation) {
+
+    this.config = new ConfiguratorConfig(
+      cConf, appNamespace,
+      artifactId, appClassName,
+      applicationName,
+      applicationVersion,
+      configString, artifactLocation.toURI());
+
+    this.remoteClientInternal = new RemoteClient(discoveryServiceClient, Constants.Service.TASK_WORKER,
+                                                 new DefaultHttpRequestConfig(false),
+                                                 String.format("%s", Constants.Gateway.INTERNAL_API_VERSION_3));
+  }
+
+  @Override
+  public ListenableFuture<ConfigResponse> config() {
+    SettableFuture<ConfigResponse> result = SettableFuture.create();
+    String jsonResponse = "";
+
+    try {
+      String param = GSON.toJson(config);
+      RunnableTaskRequest req = new RunnableTaskRequest(ConfiguratorTask.class.getName(), param);
+      String reqBody = GSON.toJson(req);
+
+      HttpRequest.Builder requestBuilder = remoteClientInternal.requestBuilder(
+        HttpMethod.POST, String.format("/worker/run")).withBody(reqBody);
+      HttpResponse response = remoteClientInternal.execute(requestBuilder.build());
+
+      jsonResponse = response.getResponseBodyAsString();
+    } catch (Exception ex) {
+      LOG.warn(String.format("Exception caught during RemoteConfigurator.config, got json response: %s", jsonResponse),
```

Comment:
So just a log?? Should it returns a failed future?

Reply:
done